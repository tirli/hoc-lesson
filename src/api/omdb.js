const baseUrl = 'http://www.omdbapi.com/?';

export function search(title) {
  return fetch(`${baseUrl}s=${title}&r=json`)
    .then((res) => res.json())
    .then((result) => result.Search.map((item) => ({
      ...item,
      Poster: item.Poster === 'N/A' ? null : item.Poster,
    })))
  ;
}

export function getById(id, plotType = 'short') {
  return fetch(`${baseUrl}i=${id}&r=json&plot=${plotType}`)
    .then((res) => res.json());
};
