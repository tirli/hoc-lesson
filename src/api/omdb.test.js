import React from 'react';
import { search, getById } from './omdb';
import searchFixture from './fixtures';

describe('OMDB api', () => {
  describe('search', () => {
    it('makes a request with correct args', () => {
      window.fetch = jest.fn((url) => Promise.resolve());
      search('hello');
      expect(window.fetch).toHaveBeenCalledTimes(1);
      expect(window.fetch).toHaveBeenCalledWith('http://www.omdbapi.com/?s=hello&r=json');
    });

    it('transforms data to correct form', async () => {
      const searchFnStub = url => Promise.resolve({ json: () => searchFixture });
      window.fetch = jest.fn(searchFnStub);
      const result = await search('smth')
      expect(result).toMatchSnapshot();
    });
  })
})

