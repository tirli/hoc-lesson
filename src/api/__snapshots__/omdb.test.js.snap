exports[`OMDB api search transforms data to correct form 1`] = `
Array [
  Object {
    "Poster": "https://images-na.ssl-images-amazon.com/images/M/MV5BMjMzMDQwMzM2M15BMl5BanBnXkFtZTcwMzA1OTg1OQ@@._V1_SX300.jpg",
    "Title": "The Test",
    "Type": "movie",
    "Year": "2013",
    "imdbID": "tt2616114",
  },
  Object {
    "Poster": null,
    "Title": "Sound Test for Blackmail",
    "Type": "movie",
    "Year": "1929",
    "imdbID": "tt0249159",
  },
]
`;
